Snowflake.Data.Tests/IntegrationTests/SFDbDataReaderGetEnumeratorIT.cs

```diff
@@ -0,0 +1,180 @@
+        {
+            using (var conn = CreateAndOpenConnection())
+            {
+                CreateAndPopulateTestTable(conn, out var cmd);
+
+                string selectCommandText = $"select * from {TableName}";
+                IDbCommand selectCmd = conn.CreateCommand();
+                selectCmd.CommandText = selectCommandText;
+                DbDataReader reader = selectCmd.ExecuteReader() as DbDataReader;
+
+                var dataRecords = reader.Cast<DbDataRecord>().ToList();
+                Assert.AreEqual(3, dataRecords.Count);
+
+                reader.Close();
+
+                DropTestTableAndCloseConnection(conn);
+            }
+        }
+
+        [Test]
+        public void TestGetEnumeratorForEachShouldNotEnterWhenResultsIsEmpty()
+        {
+            using (var conn = CreateAndOpenConnection())
+            {
+                CreateAndPopulateTestTable(conn, out var cmd);
+
+                string selectCommandText = $"select * from {TableName} WHERE cola > 10";
+                IDbCommand selectCmd = conn.CreateCommand();
+                selectCmd.CommandText = selectCommandText;
+                DbDataReader reader = selectCmd.ExecuteReader() as DbDataReader;
+
+                foreach (var record in reader)
+                {
+                    Assert.Fail("Should not enter when results is empty");
+                }
+
+                reader.Close();
+                DropTestTableAndCloseConnection(conn);
+            }
+        }
+
+        [Test]
+        public void TestGetEnumeratorShouldThrowNonSupportedExceptionWhenReset()
+        {
+            using (var conn = CreateAndOpenConnection())
+            {
+                CreateAndPopulateTestTable(conn, out var cmd);
+
+                string selectCommandText = $"select * from {TableName}";
+                IDbCommand selectCmd = conn.CreateCommand();
+                selectCmd.CommandText = selectCommandText;
+                DbDataReader reader = selectCmd.ExecuteReader() as DbDataReader;
+
+                var enumerator = reader.GetEnumerator();
+                Assert.IsTrue(enumerator.MoveNext());
+
+                Assert.Throws<NotSupportedException>(() => enumerator.Reset());
+
+                reader.Close();
+
+                DropTestTableAndCloseConnection(conn);
+            }
+        }
+
+        private void DropTestTableAndCloseConnection(DbConnection conn)
+        {
+            IDbCommand cmd = conn.CreateCommand();
+            cmd.CommandText = $"drop table if exists {TableName}";
+            var count = cmd.ExecuteNonQuery();
+            Assert.AreEqual(0, count);
+
+            CloseConnection(conn);
+        }
+
+        private void CreateAndPopulateTestTable(DbConnection conn, out IDbCommand cmd)
+        {
+            CreateOrReplaceTable(conn, TableName, new []{"cola NUMBER"});
+
+            cmd = conn.CreateCommand();
+
+            string insertCommand = $"insert into {TableName} values (1),(2),(3)";
```

Comment:
maybe we could change it for instance to 3, 5, 8?
I would like just avoid the situation that a test by coincidence could check the order of rows instead of their values.

Reply:
Got it, changed!!! Thank you!!!